### PR TITLE
feat!: support multi-line custom install-commands

### LINKED ISSUE #10: install-command does not work if the install-command contains apostrophe `"`
The input `install-command` of the `nx-cloud-main.yml@v0.2` workflow does fail when the `install-command` input contains aphostophe. 

Error Message:
```
Run if [ -n "echo "Test"
... 
/home/runner/work/_temp/f144c54f-e9f8-48bd-9d58-abcdef123.sh: line 1: [: too many arguments
```

e.g. in your ci.yaml: 
```yaml
jobs:
  main:
    name: Nx Cloud - Main Job
    uses: nrwl/ci/.github/workflows/nx-cloud-main.yml@v0.2
    with:
      node-version: ${{ github.env.NODE_VERSION }}
      install-command: |
        echo "Test"
        sudo apt-get upgrade -y
        npm ci
        ...
```

The problematic code line seems to be: https://github.com/nrwl/ci/blob/c29f238454e5782c59e60de1143f7593346e04f1/.github/workflows/nx-cloud-main.yml#L137

Apostrophe inside the `install-command` ends the if statement string and leads to unpredictable behavior.

### PR DESCRIPTION
**Pull request by WolfSoko**
- add missing `echo` command to fix the detection of package-manger for npm
- fix install-command does not work if the install-command contains apostrophe

Fixes #10

### CODE REVIEW

**Review by JamesHenry — CHANGES_REQUESTED** [review]
Thanks @WolfSoko, I just merged in some integration tests covering all package managers we support and I happened to note the missing echo on there as well so I've already applied it.

Looks like I was able to rebase your branch successfully via the Github UI, so please could you pick one of the integration tests and add the custom install command case that this is fixing so we know it is covered?

**Comment by WolfSoko** [comment]
@JamesHenry  
Wow, I just do don't get how you pushed to my forks main branch. Is this the way git works?  
Anyway good point to add tests for it! 
I added an integration test for the install-command. Please approve the workflows so that they can run.

**Comment by JamesHenry** [comment]
@WolfSoko Please pick any one of the existing workflows and an the custom command in there, I don't think we need an additional workflow just for this

**Comment by WolfSoko** [comment]
@JamesHenry 
I thought about adapting an existing worflow but decided against it.
 
Reason is, when I adapt the e.g. yarn integration test, there would be no more yarn installation anymore from the workflows default behavior and the test of that behavior is lost. 

Or do you see a different way to integrate it into a existing integration test?

**Comment by WolfSoko** [comment]
@JamesHenry can you have a look on my response again? Would be nice to get this merged.

**Comment by JamesHenry** [comment]
@WolfSoko Yeah fair enough, that offers better coverage - please also leverage the custom install command in the agent in your new workflow given that is powered separately

**Review by JamesHenry — CHANGES_REQUESTED** [review]
I think the input should also now be updated to `install-commands` to reflect that fact that it now aligns with the other multi-line string inputs and is now longer limited to a single line.

The docs in the README also need to be updated accordingly - both the usage and the descriptions of the inputs

**Comment by WolfSoko** [comment]
@JamesHenry did what you said 🙂

**Comment by WolfSoko** [comment]
@JamesHenry don't know why the runners are not picking up my commands to check  if the install-commands did run. 
There is always a syntax error.  
Maybe the way the parallel command is parsed makes it invalid. 
Any idea?

**Comment by WolfSoko** [comment]
> @JamesHenry don't know why the runners are not picking up my commands to check if the install-commands did run. There is always a syntax error. Maybe the way the parallel command is parsed makes it invalid. Any idea?

I tried hard to just use shell script to assert file existence but there was always a syntax error. Don't know the limitations of the parallel-commands and parallel-commands-on-agents and if there can't be shell scripts in there. 
But I found a way to run the file exists checks with node. Please have a look.

**Comment by WolfSoko** [comment]
@JamesHenry can you have a look again?

File: README.md [hunk 1]
```diff
@@ -139,11 +139,10 @@ jobs:
...
     pnpm-version: ""
 
-    # [OPTIONAL] If you want to provide a specific install command to use when installing dependencies
-    # you can do that here. If you do not specify one, it will check for the presence of a yarn.lock
-    # file in your repo, and if it finds one, run `yarn install --frozen-lockfile`. Otherwise it will
-    # run `npm ci`.
-    install-command: ""
+    # [OPTIONAL] If you want to provide specific install commands to use when installing dependencies
+    # you can do that here. If you do not specify one, it will check for the configured package manager
+    # (yarn, pnpm, or npm are currenctly supported) and run `npm ci` or `yarn install --frozen-lockfile`.
```
Comment (JamesHenry) [inline README.md]: It was already wonky but given this being changed please can we fix the fact it says yarn, pnpm and npm are supported but then only gives examples of npm and yarn install defaults

Comment (WolfSoko) [reply README.md]: @JamesHenry what do you think? Should I maybe put the information about supported package managers elsewhere?
I mean this install-commands are basically there to overwrite the default behavior. 
If you look for the description of the default behavior I would not search here for it.
Please check my last change for it.

**Comment by JamesHenry** [comment]
LGTM @WolfSoko, the final thing is to please update the version in the package.json to be `0.4.0` so that it gets published (tagged) appropriately post-merge

**Comment by WolfSoko** [comment]
@JamesHenry version bump to 0.4.0 done

**Review by JamesHenry — APPROVED** [review]

**Comment by JamesHenry** [comment]
Thank you @WolfSoko!

**Pull request merged**